components/gsma-authenticators/src/main/java/com/wso2telco/gsma/authenticators/dao/impl/AttributeConfigDAOimpl.java

```diff
@@ -0,0 +1,332 @@
+        Connection connection = null;
+        PreparedStatement preparedStatement = null;
+        ResultSet resultSet = null;
+
+
+        StringBuilder sqlBuilder = new StringBuilder();
+        sqlBuilder.append("INSERT INTO ");
+        sqlBuilder.append(TableName.USER_CONSENT + " (");
+        sqlBuilder.append("consent_id,msisdn,expire_time,consent_status ) ");
+        sqlBuilder.append("VALUES ");
+        sqlBuilder.append("(");
+        sqlBuilder.append("?,?,?,? )");
+
+        try {
+            connection = getConnectDBConnection();
+            preparedStatement = connection.prepareStatement(sqlBuilder.toString());
+            for (UserConsentHistory userConsentHistory1 : userConsentHistory) {
+
+                preparedStatement.setInt(1, userConsentHistory1.getConsentId());
+                preparedStatement.setString(2, userConsentHistory1.getMsisdn());
+                preparedStatement.setString(3, userConsentHistory1.getConsentExpireTime());
+                preparedStatement.setBoolean(4, Boolean.parseBoolean(userConsentHistory1.getConsentStatus()) );
+                preparedStatement.addBatch();
+            }
+
+            preparedStatement.executeBatch();
+
+        } catch (SQLException e) {
+            throw new SQLException(ERR_MSG,
+                    e);
+        } finally {
+            IdentityDatabaseUtil.closeAllConnections(connection, resultSet, preparedStatement);
+        }
+
+    }
+
+    public String getSPConfigValue(String operator, String clientID, String key)
+            throws SQLException, NamingException {
+        Connection connection = null;
+        PreparedStatement preparedStatement = null;
+        ResultSet resultSet = null;
+        String result = null;
+
+        StringBuilder sqlBuilder = new StringBuilder();
+        sqlBuilder.append(SELECT);
+        sqlBuilder.append(" sp_config.config_value ");
+        sqlBuilder.append(FROM);
+        sqlBuilder.append(TableName.SP_CONFIGURATION + " sp_config");
+        sqlBuilder.append(" where");
+        sqlBuilder.append(" sp_config.client_id=? ");
+        sqlBuilder.append(" AND sp_config.operator=? ");
+        sqlBuilder.append(" AND sp_config.config_key=? ");
+
+        try {
+            connection = getConnectDBConnection();
+            preparedStatement = connection.prepareStatement(sqlBuilder.toString());
+            preparedStatement.setString(1, clientID);
+            preparedStatement.setString(2, operator);
+            preparedStatement.setString(3, key);
+
+            resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                result = resultSet.getString("config_value");
+            }
+        } catch (SQLException e) {
+            throw new SQLException(ERR_MSG,
+                    e);
+        } finally {
+            IdentityDatabaseUtil.closeAllConnections(connection, resultSet, preparedStatement);
```

Comment:
Can remove the resource closing method by using try with resources feature. (See the [example](https://stackoverflow.com/questions/8066501/how-should-i-use-try-with-resources-with-jdbc))